R/net.dat.R

```diff
@@ -0,0 +1,385 @@
+#'        a logical vector to subset the desired `item`
+#' @param value new value to be attributed in the `set_` functions
+#' @return a vector or a list of vector for `get_` functions. And the Master
+#'         list object for `set_` and `add_` functions
+#'
+#' @name dat_get_set
+NULL
+
+#' @rdname dat_get_set
+#' @export
+get_attr_list <- function(dat, item = NULL) {
+  if (is.null(item)) {
+    out <- dat[["attr"]]
+
+  } else {
+    missing_item <- setdiff(item, names(dat[["attr"]]))
+    if (length(missing_item) > 0)
+      stop(paste("There is no attributes called",
+                 paste(missing_item, collapse = ", "),
+                 "in the attributes list of the Master list object (dat)"))
+
+    out <- dat[["attr"]][item]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+get_attr <- function(dat, item, indexes = NULL) {
+  if (!item %in% names(dat[["attr"]]))
+      stop(paste("There is no attribute called", item,
+                 "in the attributes list of the Master list object (dat)"))
+
+  if (is.null(indexes)) {
+    out <- dat[["attr"]][[item]]
+
+  } else {
+    if (is.logical(indexes)) {
+      if (length(indexes) != length(dat[["attr"]][[item]]))
+        stop("(logical) `indexes` has to have a length equal to the number of
+              nodes in the network")
+    } else if(is.numeric(indexes)) {
+      if (any(indexes > length(dat[["attr"]][[item]])))
+        stop("Some (numeric) `indexes` are larger than the number of nodes in
+              the network")
+    } else {
+      stop("`indexes` must be logical, numeric, or NULL")
+    }
+
+    out <- dat[["attr"]][[item]][indexes]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+add_attr <- function(dat, item) {
+  if (item %in% names(dat[["attr"]]))
+    stop(paste0("Cannot create the attribute '", item,
+               "': exists already"))
+
+  dat[["attr"]][[item]] <- rep(NA, length(dat$attr$active))
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+set_attr <- function(dat, item, value) {
+  if (!item %in% names(dat[["attr"]])) {
+    message(paste0("Creating attribute: '", item,
+               "' in the attributes list of the Master list object (dat)"))
+
+
+    dat <- add_attr(dat, item)
+  }
+
+  if (length(value) != length(dat$attr$active))
+    stop(paste0(
+      "When trying to edit the ", `item`, " nodale attribute: The size",
+       " of the `value` vector is not equal to the number of node in
+       the network. Expected: ", length(dat$attr$active), ", given : ",
+       length(value)))
+
+  dat[["attr"]][[item]] <- value
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+get_epi_list <- function(dat, item = NULL) {
+  if (is.null(item)) {
+    out <- dat[["epi"]]
+
+  } else {
+    missing_item <- setdiff(item,names(dat[["epi"]]))
+    if (length(missing_item) > 0)
+      stop(paste("There is no Epidemic output called",
+                 paste(missing_item, collapse = ", "),
```

Comment:
maybe let's call it epi output instead of Epidemic output

Reply:
done